### ISSUE TITLE
Unexpected null values

### ISSUE THREAD

**Issue by dkuppitz** [thinkaurelius/titan#355]
One of our clients has a problem with unexpected null values. Unfortunately I cannot reproduce this behaviour in s small sample graph, but here's how it looks like:

```
g.V[1..10].map()
<snip>
==>{a_property=null, custom_id=-7930742963208686808, ...}
```

Then, in the same graph (and even the same Gremlin REPL session) the following query:

```
g.v(203003840).map()
==>a_property=a_string_value
==>custom_id=-7930742963208686808
...
```

The custom id is definitely unique, so it's definitely the same vertex. Some queries return null values for a_property, some queries return the real (expected) values. It's not only this single vertex, this behaviour is seen in a lot of (random) vertices, too. We can't tell for sure, but it seems like only this a_property is affected. Unexpected null values were not seen on other properties.

System: Cassandra + ElasticSearch

**Comment by mbroecheler** [comment]
When you do the first query after the second one, do you still get the null
value?

On Thu, Aug 22, 2013 at 1:17 PM, Daniel Kuppitz notifications@github.comwrote:

> One of our clients has a problem with unexpected null values.
> Unfortunately I cannot reproduce this behaviour in s small sample graph,
> but here's how it looks like:
> 
> g.V[1..10].map()
> <snip>
> ==>{a_property=null, custom_id=-7930742963208686808, ...}
> 
> Then, in the same graph (and even the same Gremlin REPL session) the
> following query:
> 
> g.v(203003840).map()
> ==>a_property=a_string_value
> ==>custom_id=-7930742963208686808
> ...
> 
> The custom id is definitely unique, so it's definitely the same vertex.
> Some queries return null values for a_property, some queries return the
> real (expected) values. It's not only this single vertex, this behaviour is
> seen in a lot of (random) vertices, too. We can't tell for sure, but it
> seems like only this a_property is affected. Unexpected null values were
> not seen on other properties.
> 
> System: Cassandra + ElasticSearch
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/thinkaurelius/titan/issues/355
> .

## 

Matthias Broecheler
http://www.matthiasb.com

**Comment by okram** [comment]
We noticed this with another client when bulk loading data into Titan from Faunus and the property wasn't defined prior to bulk loading. Titan gets into a weird state where null values are returned though the data in Cassandra is there and not corrupt. moreover, as you note, sometimes it returns the value okay and sometimes not --- seemingly random.

**Comment by dkuppitz** [comment]
When I execute the second query first, it's the same result. `g.v(id).map()` has the expected value, `gV[0..10].map()` has the null value.

**Comment by dkuppitz** [comment]
> and the property wasn't defined prior to bulk loading

I can confirm, that this behaviour can be avoided by defining all types prior to bulk loading. It only seems to be a problem when you forget a type definition.

**Comment by mbroecheler** [comment]
I think what I will add for Titan 0.4.0 is a switch that disables automatic type definition when batch-loading is enabled.

**Comment by mbroecheler** [comment]
Fixed by #289 as described in previous comment.
